go/vt/schemadiff/table.go

```diff
@@ -799,3 +841,228 @@ func (c *CreateTableEntity) Drop() EntityDiff {
 	}
 	return &DropTableEntityDiff{from: c, dropTable: dropTable}
 }
+
+// apply attempts to apply an ALTER TABLE diff onto this entity's table definition.
+// supported modifications are only those created by schemadiff's Diff() function.
+func (c *CreateTableEntity) apply(diff *AlterTableEntityDiff) error {
+	if spec := diff.alterTable.PartitionSpec; spec != nil {
+		switch {
+		case spec.Action == sqlparser.RemoveAction && spec.IsAll:
+			// Remove partitioning
+			c.TableSpec.PartitionOption = nil
+		default:
+			return errors.Wrapf(ErrUnsupportedApplyOperation, "%v", sqlparser.String(spec))
+		}
+	}
+	if diff.alterTable.PartitionOption != nil {
+		// Specify new spec:
+		c.CreateTable.TableSpec.PartitionOption = diff.alterTable.PartitionOption
+	}
+	// reorderColumn attempts to reorder column that is right now in position 'colIndex',
+	// based on its FIRST or AFTER specs (if any)
+	reorderColumn := func(colIndex int, first bool, after *sqlparser.ColName) error {
+		var newCols []*sqlparser.ColumnDefinition // nil
+		col := c.TableSpec.Columns[colIndex]
+		switch {
+		case first:
+			newCols = append(newCols, col)
+			newCols = append(newCols, c.TableSpec.Columns[0:colIndex]...)
+			newCols = append(newCols, c.TableSpec.Columns[colIndex+1:]...)
+		case after != nil:
+			afterColFound := false
+			// look for the AFTER column; it has to exist!
+			for a, afterCol := range c.TableSpec.Columns {
+				if afterCol.Name.String() == after.Name.String() {
+					if colIndex < a {
+						// moving column i to the right
+						newCols = append(newCols, c.TableSpec.Columns[0:colIndex]...)
+						newCols = append(newCols, c.TableSpec.Columns[colIndex+1:a+1]...)
+						newCols = append(newCols, col)
+						newCols = append(newCols, c.TableSpec.Columns[a+1:]...)
+					} else {
+						// moving column i to the left
+						newCols = append(newCols, c.TableSpec.Columns[0:a+1]...)
+						newCols = append(newCols, col)
+						newCols = append(newCols, c.TableSpec.Columns[a+1:colIndex]...)
+						newCols = append(newCols, c.TableSpec.Columns[colIndex+1:]...)
+					}
+					afterColFound = true
+					break
+				}
+			}
+			if !afterColFound {
+				return errors.Wrapf(ErrApplyColumnNotFound, "%v", after.Name.String())
```

Comment:
Can this be simplified to `errors.Wrap(ErrApplyColumnNotFound, after.Name.String())`? And for all the other further cases down in this file?

Reply:
done